application/controllers/HistoryController.php

```diff
@@ -49,11 +50,15 @@ public function indexAction()
             ]
         );
         $viewModeSwitcher = $this->createViewModeSwitcher($paginationControl, $limitControl, true);
-        $searchBar = $this->createSearchBar($history, [
+        $preserveParams = [
             $limitControl->getLimitParam(),
             $sortControl->getSortParam(),
             $viewModeSwitcher->getViewModeParam()
-        ]);
+        ];
+
+
+        $searchBar = $this->createSearchBar($history, $preserveParams);
+        $url = $url->onlyWith($preserveParams);
```

Comment:
You should put that to the same place where the query string is handled. Well, why not even put it right above where it's required. (Around line no. 103) You might even move the initialization of `$url` there as well. To shorten everything a bit:

```
$url = Url::fromRequest()->onlyWith...
$url->setQueryString(QueryString::render(...
```